infra-gen2/tool/deploy_gen2.dart

```diff
@@ -0,0 +1,391 @@
+#!/usr/bin/env dart
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:path/path.dart' as p;
+
+class AmplifyIntegCategory {
+  const AmplifyIntegCategory({
+    required this.category,
+    required this.backends,
+    required this.defaultOutput,
+    this.sharedOutputs = const [],
+  });
+
+  /// This is the category of the integration group
+  final Category category;
+
+  /// This is the list of backends for the category
+  final List<AmplifyBackend> backends;
+
+  /// This is the default output path for the category
+  final String defaultOutput;
+
+  /// This is the list of shared output paths for the category
+  final List<String> sharedOutputs;
+}
+
+class AmplifyBackend {
```

Comment:
Q: Could this info could be created by iterating over the infra-gen2/backends dir? Is there an advantage to listing it out?

Reply:
Yeah, that was my first approach. I went this route because there were small nuances like naming requirements differing between `bucketName` & `bucketIdentifier` and specifying them explicitly made it easier to reason about. 

Also, this lets us create/reuse back ends not bound to one particular plugin category.  

Now, it certainly could be different/improved, open to suggestions.